osu.Game/Database/LegacyArchiveExporter.cs

```diff
@@ -0,0 +1,85 @@
+        where TModel : RealmObject, IHasNamedFiles, IHasGuidPrimaryKey
+    {
+        protected LegacyArchiveExporter(Storage storage, RealmAccess realm)
+            : base(storage, realm)
+        {
+        }
+
+        protected override void ExportToStream(TModel model, Stream outputStream, ProgressNotification notification, CancellationToken cancellationToken = default)
+            => exportZipArchive(model, outputStream, notification, cancellationToken);
+
+        /// <summary>
+        /// Exports an item to Stream as a legacy (.zip based) package.
+        /// </summary>
+        /// <param name="model">The model will be exported.</param>
+        /// <param name="outputStream">The output stream to export to.</param>
+        /// <param name="notification">The notification will displayed to the user</param>
+        /// <param name="cancellationToken">The Cancellation token that can cancel the exporting.</param>
+        private void exportZipArchive(TModel model, Stream outputStream, ProgressNotification notification, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                using var writer = new ZipWriter(outputStream, new ZipWriterOptions(CompressionType.Deflate));
+
+                float i = 0;
+                bool fileMissing = false;
+
+                foreach (var file in model.Files)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    using (var stream = UserFileStorage.GetStream(file.File.GetStoragePath()))
+                    {
+                        // Sometimes we cannot find the file(probably deleted by the user), so we handle this and post a error.
+                        if (stream == null)
+                        {
+                            // Only pop up once to prevent spam.
+                            if (!fileMissing)
+                            {
+                                PostNotification?.Invoke(new SimpleErrorNotification
+                                {
+                                    Text = "Some of your files are missing, they will not be included in the archive"
+                                });
+                                fileMissing = true;
+                            }
+                        }
+                        else
+                        {
+                            writer.Write(file.Filename, stream);
+                        }
+                    }
+
+                    i++;
+                    notification.Progress = i / model.Files.Count();
+                    notification.Text = $"Exporting... ({i}/{model.Files.Count()})";
+                }
+            }
+            catch (ObjectDisposedException)
+            {
+                // outputStream may close before writing when request cancel.
+                if (cancellationToken.IsCancellationRequested)
+                    return;
+
+                throw;
+            }
```

Comment:
I don't even know what is happening here.

Reply:
It looks like if the user cancel
https://github.com/ppy/osu/pull/21308/files#diff-6a0e3884e01dc22403790923a038a2cdda6d02f121b621682c78bc3b12b7c833R100
it will run out of range first and dispose the stream and cause `ObjectDisposedExcption`
Because it can easily happen during the cancellation process, I choose to catch and only return when the user cancels